Assets/MixedRealityToolkit.Providers/UnityAR/UnityARCameraSettings.cs

```diff
@@ -0,0 +1,255 @@
+            if (isInitialized) { return; }
+
+            FindARFoundationComponents();
+
+            if (arSessionObject == null)
+            {
+                arSessionObject = new GameObject("AR Session");
+                arSessionObject.transform.parent = null;
+            }
+            arSession = arSessionObject.EnsureComponent<ARSession>();
+            arInputManager = arSessionObject.EnsureComponent<ARInputManager>();
+
+            if (arSessionOriginObject == null)
+            {
+                arSessionOriginObject = MixedRealityPlayspace.Transform.gameObject;
+            }
+            CameraCache.Main.transform.parent = arSessionOriginObject.transform;
+
+            arSessionOrigin = arSessionOriginObject.EnsureComponent<ARSessionOrigin>();
+            arSessionOrigin.camera = CameraCache.Main;
+
+            GameObject cameraObject = arSessionOrigin.camera.gameObject;
+
+            arCameraManager = cameraObject.EnsureComponent<ARCameraManager>();
+            arCameraBackground = cameraObject.EnsureComponent<ARCameraBackground>();
+
+            trackedPoseDriver = cameraObject.EnsureComponent<TrackedPoseDriver>();
+
+            TrackedPoseDriver.TrackedPose poseSource;
+            TrackedPoseDriver.TrackingType trackingType;
+            TrackedPoseDriver.UpdateType updateType;
+
+            if (SettingsProfile != null)
+            {
+                // Read settings to be applied to the camera.
+                poseSource = SettingsProfile.PoseSource;
+                trackingType = SettingsProfile.TrackingType;
+                updateType = SettingsProfile.UpdateType;
+            }
+            else
+            {
+                Debug.LogWarning("A profile was not specified for the XR Camera Settings provider.\nApplying Microsoft Mixed Reality Toolkit default options.");
+                // Use default settings.
+                poseSource = TrackedPoseDriver.TrackedPose.ColorCamera;
+                trackingType = TrackedPoseDriver.TrackingType.RotationAndPosition;
+                updateType = TrackedPoseDriver.UpdateType.UpdateAndBeforeRender;
+            }
+
+            trackedPoseDriver.SetPoseSource(
+                TrackedPoseDriver.DeviceType.GenericXRDevice,
+                poseSource);
+            trackedPoseDriver.trackingType = trackingType;
+            trackedPoseDriver.updateType = updateType;
+            trackedPoseDriver.UseRelativeTransform = false;
+
+            isInitialized = true;
+        }
+
+        /// <summary>
+        /// Uninitialize and clean up AR Foundation components.
+        /// </summary>
+        private void UninitializeARFoundation()
+        {
+            if (!isInitialized) { return; }
+
+            if (!preExistingArSessionOriginObject &&
+                (arSessionOriginObject != null))
+            {
+                if (Application.isEditor && !Application.isPlaying)
+                {
+                    Object.DestroyImmediate(trackedPoseDriver);
```

Comment:
GameObjectExtensions.DestroyGameObject? or Object.Destroy to handle the branching?

Reply:
trackedPoseDriver is not a GameObject (it's a script). we will need a new extension to handle

Reply:
yea sorry that is what I meant by Object.DestroyObject() as an extension in ObjectExtensions.cs

Reply:
ahh. ok. i'll double check :)